### ISSUE TITLE
Exception thrown if UTF-8 characters present in source code

### ISSUE THREAD

**Issue by jshiell** [jshiell/checkstyle-idea#84]
From Anders Smestad:
```
It crashes if you have non-ascii characters in the source code. For instance, we have a java class named no.mat.Blåbærsyltetøy and run the checkstyle plugin on it, it says: "Got an exception - Unexpected character 0xfffd in identifier (xx:xx)

even if the file is utf-8 encoded and java/intellij compiles it just fine)
```

**Comment by hdavid** [comment]
Hello,

i have a similar, if not the same problem :
![untitled](https://cloud.githubusercontent.com/assets/566554/6753095/efac90d2-cf13-11e4-96d1-0570206dd34b.png)

it works with maven/checkstyle. but not in intellij. i am happy to provide the source file and pom and config files in a private email

regards,

henri

**Comment by jshiell** [comment]
Superb news (for me, appreciate it's pretty inconvenient for you :smile:) - I'd love to get this sorted, so if you could please send them to james+csidea AT infernus org I'd be most grateful!

**Comment by jshiell** [comment]
Thanks for the files - I'll hopefully get on this during the weekend!

**Comment by jshiell** [comment]
I'm still not able to reproduce it :disappointed: I've tried the following combinations, all on OS X 10.10.2 with language setup of English UK (primary); English.

IntelliJ 14.0.3 (JRE 1.6.0_65, default encoding MacRoman); project SDK 1.6.0_65 (MacRoman)
IntelliJ 14.1.1 141.178 (JRE 1.8.0_40 custom build by Jetbrains; default UTF-8); project SDK 1.6.0_65 (MacRoman)
IntelliJ 14.1.1 141.178 (JRE 1.8.0_40 custom build by Jetbrains; default UTF-8); project SDK 1.8.0_31 (UTF-8)

I don't think it's an OS issue, as the original reporter was running OS X 10.9 (as was I, at the time). But I'm not sure where the language discrepency is creeping in. If you can offer any further information on your OS or JDK language/encodings, or any thoughts as to where else my setup might differ, I'd be most grateful.

For the record, I was using the following to confirm the encodings of my JDKs:
```
import java.nio.charset.Charset;
import java.io.InputStreamReader;
import java.io.ByteArrayInputStream;

public class Encodings {
    public static void main(final String[] args) {
        System.out.println("File encoding = " + System.getProperty("file.encoding"));
        System.out.println("Charset = " + Charset.defaultCharset());
        System.out.println("InputStreamReader = " + new InputStreamReader(new ByteArrayInputStream(new byte[] {'a'})).getEncoding());
    }
}
```

**Comment by hdavid** [comment]
Hello,

we are on windows 7 64bit, with stock intellij 14.0.x.
i believe my regional setting are us or uk, but i dont think that would
make a difference.

Our source is in utf-8. maybe something somewhere loads the source using
platform encoding ?

**Comment by jshiell** [comment]
You're almost certainly correct on that. I'm just very confused as to why it's not showing up with the MacRoman JVM.

In any case, my next step will be to manually review the code and see if I can spot anything suspicious.

**Comment by jshiell** [comment]
I've had a good (if belated, sorry) poke around, and the only thing I've found is that the file writes weren't specifying the encoding. I've changed this to specify UTF-8, but as I can't reproduce this I've no idea if it makes any difference.

If you have a moment, I'd be very grateful if you could have a try with [this test build](https://dl.dropboxusercontent.com/u/5423904/checkstyle-idea-4.14.2.zip) and see if it makes any difference for you. Thank you!

**Comment by hdavid** [comment]
i had a quick look, and it seemed to work !

**Comment by jshiell** [comment]
Thanks very much for your help! I've released this as 4.14.2 - if the problem reoccurs, please open this and I'll have another look :smile:
